components/infobox/wikis/rainbowsix/infobox_league_custom.lua

```diff
@@ -0,0 +1,273 @@
+	local league = League(frame)
+	_league = league
+	_args = _league.args
+
+	league.createWidgetInjector = CustomLeague.createWidgetInjector
+	league.defineCustomPageVariables = CustomLeague.defineCustomPageVariables
+	league.addToLpdb = CustomLeague.addToLpdb
+
+	return league:createInfobox(frame)
+end
+
+function CustomLeague:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:addCustomCells(widgets)
+	local args = _args
+	table.insert(widgets, Cell{
+		name = 'Teams',
+		content = {(args.team_number or '') .. (args.team_slots and ('/' .. args.team_slots) or '')}
+	})
+	table.insert(widgets, Cell{
+		name = 'Game',
+		content = {CustomLeague:_createGameCell(args)}
+	})
+	table.insert(widgets, Cell{
+		name = 'Platform',
+		content = {CustomLeague:_createPlatformCell(args)}
+	})
+	table.insert(widgets, Cell{
+		name = 'Players',
+		content = {args.player_number}
+	})
+
+	return widgets
+end
+
+function CustomInjector:parse(id, widgets)
+	local args = _args
+	if id == 'customcontent' then
+		if String.isNotEmpty(args.map1) then
+			local game = String.isNotEmpty(args.game) and ('/' .. args.game) or ''
+			local maps = {}
+
+			for _, map in ipairs(_league:getAllArgsForBase(args, 'map')) do
+				table.insert(maps, tostring(CustomLeague:_createNoWrappingSpan(
+					PageLink.makeInternalLink({}, map, map .. game)
+				)))
+			end
+			table.insert(widgets, Title{name = 'Maps'})
+			table.insert(widgets, Center{content = {table.concat(maps, '&nbsp;• ')}})
+		end
+	elseif id == 'prizepool' then
+		return {
+			Cell{
+				name = 'Prize pool',
+				content = {CustomLeague:_createPrizepool()}
+			},
+		}
+	elseif id == 'liquipediatier' then
+		widgets = {
+			Cell{
+				name = 'Liquipedia Tier',
+				content = {CustomLeague:_createLiquipediaTierDisplay()},
+			}
+		}
+		if String.isNotEmpty(args.ubisofttier) and _UBISOFT_TIERS[args.ubisofttier:lower()] then
+			table.insert(widgets,
+				Cell{
+					name = 'Ubisoft Tier',
+					content = {'[['.._UBISOFT_TIERS[args.ubisofttier:lower()]..']]'},
+					classes = {'valvepremier-highlighted'}
+				}
+			)
+		end
+	end
+	return widgets
+end
+
+function CustomLeague:addToLpdb(lpdbData, args)
+	lpdbData.maps = table.concat(_league:getAllArgsForBase(args, 'map'), ';')
+
+	lpdbData.publishertier = args.ubisofttier:lower()
+	lpdbData.participantsnumber = args.player_number or args.team_number
+	lpdbData.extradata = {
+		individual = String.isNotEmpty(args.player_number) and 'true' or '',
+		startdatetext = CustomLeague:_standardiseRawDate(args.sdate or args.date),
+		enddatetext = CustomLeague:_standardiseRawDate(args.edate or args.date),
+	}
+
+	return lpdbData
+end
+
+function CustomLeague:_standardiseRawDate(dateString)
+	-- Length 7 = YYYY-MM
+	-- Length 10 = YYYY-MM-??
+	if String.isEmpty(dateString) or (#dateString ~= 7 and #dateString ~= 10) then
+		return ''
+	end
+
+	if #dateString == 7 then
+		dateString = dateString .. '-??'
+	end
+	dateString = dateString:gsub('-XX', '-??')
```

Comment:
i personally would escape the `-` even though lua doesn't interpret it as reg ex due to it being at the start of the regex string
```suggestion
	dateString = dateString:gsub('%-XX', '-??')
```